content/blog/aws-eks-managed-nodes-fargate/index.md

```diff
@@ -0,0 +1,356 @@
+
+// Export the cluster's kubeconfig.
+export const kubeconfig = cluster.kubeconfig;
+```
+
+There is even more you can do here, including digging deep into the individual EC2 instances and Auto Scaling Groups powering the node group. Needing to manage workers at this level, and connect them to the right IAM and networking services, can be tedious and difficult to get right. That's where the new AWS features come into play.
+
+## Automatically Managed Node Groups
+
+The new EKS feature [Managed Node Groups](https://aws.amazon.com/blogs/containers/eks-managed-node-groups/) simplifies the task of managing explicit pools of worker nodes, at the cost of some amount of control. The `@pulumi/eks` package already had many of these conveniences built-in but this is now an official feature of the AWS platform.
+
+Managed Node Groups will automatically scale the EC2 instances powering your cluster using an Auto Scaling Group managed by EKS. This ASG also runs the latest Amazon EKS-optimized Amazon Linux 2 AMI. This is great on one hand &mdash; because updates will be applied automatically for you &mdash; but if you want control over this you will want to manage your own node groups. Finally, security groups, IAM roles, and connecting them together is handled for you.
+
+To opt-in to using Managed Node Groups, the raw [`aws.eks.NodeGroup` building block](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/aws/eks/#NodeGroup) is available. However, we've added the `eks.Cluster.createManagedNodeGroup` function to make it easier, and to integrate with cluster provisioning.
+
+The following example creates an EKS cluster with a single Managed Node Group. It looks similar to our explicitly managed node group earlier, but a bit simpler because we can lean on EKS to configure and scale it:
+
+```typescript
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+
+// Create a VPC for our cluster.
+const vpc = new awsx.ec2.Vpc("my-vpc");
+
+// IAM roles for the node group.
+const role = new aws.iam.Role("my-cluster-ng-role", {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+        Service: "ec2.amazonaws.com",
+    }),
+});
+let counter = 0;
+for (const policyArn of [
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+]) {
+    new aws.iam.RolePolicyAttachment(`my-cluster-ng-role-policy-${counter++}`,
+        { policyArn, role },
+    );
+}
+
+// Create an EKS cluster.
+const cluster = new eks.Cluster("my-cluster", {
+    skipDefaultNodeGroup: true,
+    vpcId: vpc.vpcId,
+    publicSubnetIds: vpc.publicSubnetIds,
+    privateSubnetIds: vpc.privateSubnetIds,
+    deployDashboard: false,
+    instanceRoles: [ role ],
+});
+
+// Create a simple AWS managed node group using a cluster as input.
+const managedNodeGroup = eks.createManagedNodeGroup("my-cluster-ng", {
```

Comment:
@metral @lukehoban Do you know why `createManagedNodeGroup` is so stylistically different than `createNodeGroup`? Note that I need to pass the `cluster` as an argument (whereas with `createNodeGroup`, it's a member function on the cluster itself) and, as a result, I need to pass it twice (to get the `parent` wired up).

Reply:
> Do you know why createManagedNodeGroup is so stylistically different than createNodeGroup?

We tried to set some defaults where possible, while also inferring settings so the user doesn't have to be so explicit. But optional enough that if they want to provide full configuration they still can. Here is the defaulting we do:

https://github.com/pulumi/pulumi-eks/blob/e9d4c42036c56c7fd526ab6c7ddd85ff10d0fb17/nodejs/eks/nodegroup.ts#L638-L672

Do note that we aren't radically changing the spec's here, just simplifying it for folks where it makes sense. `aws.eks.NodeGroup` has a spec that closely resembles our `eks.NodeGroup` class, so there isn't much more reduction we can do that we haven't already incorporated.

> Note that I need to pass the cluster as an argument (whereas with createNodeGroup, it's a member function on the cluster itself) and, as a result, I need to pass it twice (to get the parent wired up).

This is a simple fix we can consolidate so we can auto-set the parent w/o having to pass it in. I opened an issue to track this. If you have any other ideas to correct the API style here, please drop a note in the issue:
https://github.com/pulumi/pulumi-eks/issues/285